src/components/event_view/event_info/EventInfo.js

```diff
@@ -1,38 +1,148 @@
 import React, { Component } from 'react';
-import PlanInfo from '../events/PlanInfo';
-import AccomInfo from '../events/AccomInfo';
-import TranInfo from '../events/TranInfo';
 import PropTypes from 'prop-types';
+import { Table } from 'react-bootstrap';
+import EventField from '../event_fields/EventField';
+import EventFieldEdit from '../event_fields/EventFieldEdit';
+
+export class EventInfo extends Component {
+  constructor(props) {
+    super(props);
+    this.state = props.state;
+    this.onEdit = this.onEdit.bind(this);
+    this.onCancel = this.onCancel.bind(this);
+    this.onSave = this.onSave.bind(this);
+  }
+
+  /**
+   * Reserves editing to just one field element at a time.
+   * @param fieldName to reserve edit mode for.
+   */
+  reserveEditMode(fieldName) {
+    for (let f in this.state) {
+      if (fieldName !== f) {
+        this.setState({
+          [f]: {
+            ...this.state[f],
+            editMode: false
+          }
+        });
+      }
+    }
+  }
+
+  /**
+   * Gets either a editable or viewable field representing a value of a
+   * plan object.
+   * @param field to be retrieved.
+   * @returns {*} either an editable or viewable field.
+   */
+  getField(field) {
+    if (field.editMode === true) {
```

Comment:
=== true is redundant

Reply:
I wouldn't say it's redundant, it's a preferential thing. I'm easy to swing either way on this notation though. Just changed it to ``` if (field.editMode) ```.